packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -550,6 +550,9 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
     if (searchText.isEmpty) {
       return null;
     }
+    if (currentHighlight != null && entries[currentHighlight!].label.toLowerCase() == searchText) {
```

Comment:
nit: The default search algorithm is just to search an item that contains the queried text. Do you think if we should just use `contains(...)` instead of exact matching? :)

Reply:
I believe this check is meant for when `currentHighlight` is set using the up/down arrow keys, which means there's always an exact match.

But using `contains(...)` here would let us skip the loop in other situations too, so that sounds great!

```suggestion
    if (currentHighlight != null && entries[currentHighlight!].label.toLowerCase().contains(searchText)) {
```

Reply:
As a side effect of using `contains(...)` searching would prioritises the current highlight. This should be better unless it breaks something.